### PR TITLE
Add OAuth2LoginSpec#securityContextRepository

### PR DESCRIPTION
**Pull request by ilgrosso**
Following discussion at #7222 , `OAuth2LoginSpec#securityContextRepository` was added.

### CODE REVIEW

**Comment by ilgrosso** [comment]
Travis CI failed because:

```
A problem occurred configuring project ':spring-security-samples-xml-gae'.
> Could not resolve all artifacts for configuration ':spring-security-samples-xml-gae:classpath'.
   > Could not resolve com.google.cloud.tools:appengine-gradle-plugin:1.3.5.
     Required by:
         project :spring-security-samples-xml-gae
      > Could not resolve com.google.cloud.tools:appengine-gradle-plugin:1.3.5.
         > Could not get resource 'https://repo.maven.apache.org/maven2/com/google/cloud/tools/appengine-gradle-plugin/1.3.5/appengine-gradle-plugin-1.3.5.pom'.
            > Could not HEAD 'https://repo.maven.apache.org/maven2/com/google/cloud/tools/appengine-gradle-plugin/1.3.5/appengine-gradle-plugin-1.3.5.pom'. Received status code 403 from server: Forbidden
```

which is clearly not related to the proposed changes.

File: config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java [hunk 1]
```diff
@@ -970,7 +970,9 @@ public ServerHttpSecurity oauth2Login(Customizer<OAuth2LoginSpec> oauth2LoginCus
 
 		private ServerOAuth2AuthorizedClientRepository authorizedClientRepository;
 
-		private ReactiveAuthenticationManager authenticationManager;
+                private ReactiveAuthenticationManager authenticationManager;
```
Comment (jzheaux) [inline config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java]: Something odd happened to the whitespace here, could you correct that please?

File: config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java [hunk 2]
```diff
@@ -993,6 +995,19 @@ public OAuth2LoginSpec authenticationManager(ReactiveAuthenticationManager authe
...
 
+		/**
+		 * The {@link ServerSecurityContextRepository} used to save the {@code Authentication}. Defaults to
+		 * {@link WebSessionServerSecurityContextRepository}.
+		 *
+		 * @since 5.2
+		 * @param securityContextRepository the repository to use
+		 * @return the {@link OAuth2LoginSpec} to continue configuring
+		 */
+		public OAuth2LoginSpec securityContextRepository(ServerSecurityContextRepository securityContextRepository) {
```
Comment (jzheaux) [inline config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java]: Would you please add a simple test to `OAuth2LoginTests` that confirms that an application-provided security context repository is correctly picked up?

File: config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java (hunk 1 above)
Comment (ilgrosso) [reply config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java]: Done, sorry for the whitespace noice

File: config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java (hunk 2 above)
Comment (ilgrosso) [reply config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java]: I modified an existing test, I hope it's fine enough.

**Comment by ilgrosso** [comment]
@jzheaux please take a look again and let me know, thx

**Comment by jzheaux** [comment]
Thanks, @ilgrosso! This is now merged into `master`.

**Comment by ilgrosso** [comment]
Cool, thx for review @jzheaux

**Pull request merged**